shell/platform/android/io/flutter/embedding/engine/systemchannels/NavigationChannel.java

```diff
@@ -57,7 +60,40 @@ public void popRoute() {
     channel.invokeMethod("popRoute", null);
   }
 
+  @TargetApi(34)
+  @RequiresApi(34)
+  public void startBackGesture(@NonNull BackEvent backEvent) {
+    Log.v(TAG, "Sending message to start back gesture");
+    channel.invokeMethod("startBackGesture", backEventToJsonMap(backEvent));
+  }
+
+  @TargetApi(34)
+  @RequiresApi(34)
+  public void updateBackGestureProgress(@NonNull BackEvent backEvent) {
+    Log.v(TAG, "Sending message to update back gesture progress");
+    channel.invokeMethod("updateBackGestureProgress", backEventToJsonMap(backEvent));
+  }
+
+  @TargetApi(34)
+  @RequiresApi(34)
+  public void cancelBackGesture() {
+    Log.v(TAG, "Sending message to cancel back gesture");
+    channel.invokeMethod("cancelBackGesture", null);
+  }
+
   public void setMethodCallHandler(@Nullable MethodChannel.MethodCallHandler handler) {
     channel.setMethodCallHandler(handler);
   }
+
+  @TargetApi(34)
+  @RequiresApi(34)
+  private Map<String, Object> backEventToJsonMap(@NonNull BackEvent backEvent) {
+    Map<String, Object> message = new HashMap<>(4);
+    message.put("touchX", backEvent.getTouchX());
+    message.put("touchY", backEvent.getTouchY());
+    message.put("progress", backEvent.getProgress());
+    message.put("swipeEdge", backEvent.getSwipeEdge() == BackEvent.EDGE_LEFT ? "left" : "right");
```

Comment:
Nit: You could just forward the Android constants to the framework and decode it on that side with a comment linking to the docs instead of using these Strings for clarity.